From f59c5269f92d59a5296cbfeeb682d42095cd88ad Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 18 Sep 2014 02:24:07 -0400
Subject: [PATCH] gstreamer1.0-plugins-bad: allow to disable libssh2

libssh2 is automatically linked to if present, this undetermined
dependency may cause build errors like:

  .../x86_64-poky-linux/4.9.0/ld: cannot find -lssh2

libssh2 isn't an oe-core recipe, so allow to disable it from
configure.

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 configure.ac |   23 +++++++++++++++++------
 1 file changed, 17 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0e95c5c..12153b4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1901,6 +1901,15 @@ AG_GST_CHECK_FEATURE(CHROMAPRINT, [chromaprint], chromaprint, [
 ])
 
 dnl *** Curl ***
+AC_ARG_ENABLE([libssh2],
+     [  --enable-libssh2		enable LIBSSH2 support @<:@default=auto@:>@],
+     [case "${enableval}" in
+       yes)  NEED_SSH2=yes ;;
+       no)   NEED_SSH2=no ;;
+       auto) NEED_SSH2=auto ;;
+       *) AC_MSG_ERROR([bad value ${enableval} for --enable-libssh2]) ;;
+     esac],[NEED_SSH2=auto])
+
 translit(dnm, m, l) AM_CONDITIONAL(USE_CURL, true)
 AG_GST_CHECK_FEATURE(CURL, [Curl plugin], curl, [
   PKG_CHECK_MODULES(CURL, libcurl >= 7.21.0, [
@@ -1915,12 +1924,14 @@ AG_GST_CHECK_FEATURE(CURL, [Curl plugin], curl, [
   ])
   AC_SUBST(CURL_CFLAGS)
   AC_SUBST(CURL_LIBS)
-  PKG_CHECK_MODULES(SSH2, libssh2 >= 1.4.3, [
-    HAVE_SSH2="yes"
-    AC_DEFINE(HAVE_SSH2, 1, [Define if libssh2 is available])
-  ], [
-    HAVE_SSH2="no"
-  ])
+  if test "x$NEED_SSH2" != "xno"; then
+    PKG_CHECK_MODULES(SSH2, libssh2 >= 1.4.3, [
+      HAVE_SSH2="yes"
+      AC_DEFINE(HAVE_SSH2, 1, [Define if libssh2 is available])
+    ], [
+      HAVE_SSH2="no"
+    ])
+  fi
   AM_CONDITIONAL(USE_SSH2, test "x$HAVE_SSH2" = "xyes")
   AC_SUBST(SSH2_CFLAGS)
   AC_SUBST(SSH2_LIBS)
-- 
1.7.9.5

